libs/backend-api7/src/transformer.ts

```diff
@@ -4,11 +4,9 @@ import { attempt, isError } from 'lodash';
 import * as typing from './typing';
 
 export class ToADC {
-  private static transformLabels(
-    labels?: ADCSDK.Labels,
-  ): Record<string, string> {
-    if (!labels) return undefined;
-    return Object.entries(labels).reduce((pv, [key, value]) => {
+  private static transformLabels(labels?: ADCSDK.Labels): ADCSDK.Labels {
+    if (!labels) return {} as ADCSDK.Labels;
```

Comment:
Keep `undefined`.

The `{}` is clearly not equal to `undefined`. The `undefined` will be removed from the final output by `recursiveOmitUndefined`, whereas your modification cannot achieve this.